cli/cycpp.py

```diff
@@ -1547,8 +1563,11 @@ def _buildschema(self, cpptype, schematype=None, uitype=None, names=None):
             impl += '</element>'
         elif t == 'std::map':
             name = 'map'
-            if names[0] is not None:
-                name = names[0]
+            if isinstance(names[0], STRING_TYPES):
+                names[0] = [names[0], None]
```

Comment:
Hi everyone,

The show stopper here for me is compatibility with the rest of the Cyclus
project, namely Cycic/Cyclist. I suggest we bring this discussion up on the
dev call today and do our best to figure out the right path forward for
everyone.

Cheers,
Matt


On Wed, May 6, 2015 at 9:45 AM, Anthony Scopatz <notifications@github.com>
wrote:

> In cli/cycpp.py
> <https://github.com/cyclus/cyclus/pull/1143#discussion_r29764732>:
>
> > @@ -1547,8 +1563,11 @@ def _buildschema(self, cpptype, schematype=None, uitype=None, names=None):
> >              impl += '</element>'
> >          elif t == 'std::map':
> >              name = 'map'
> > -            if names[0] is not None:
> > -                name = names[0]
> > +            if isinstance(names[0], STRING_TYPES):
> > +                names[0] = [names[0], None]
>
> put in a fix
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1143/files#r29764732>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192